### DISCUSSION TITLE
hide the file list

### DISCUSSION THREAD

**Discussion by MuZhou233** [spencerwooo/onedrive-vercel-index#492] [Q&A] [answered]
Hi there,

I'm interesting in how to hide the file list by a specific file just like the `.password` file. In my opinion, a list hided folder should *seems* like an empty folder(or just shows an `README.md` file).

Could I know how to reach that? Or if someone will make this function a mainline feature.

**Comment by spencerwooo** [comment]
It's currently not available, and do you mean that you want to collapse the file list?

**Reply by MuZhou233** [reply]
No, What I want is the hidden folder just shows an README file in, but if I visit the file or un-hidden folder in that folder, It works normal.

By the way, I used reach that by a `<style>` tag with CSS in README file in other project. but in this project, the file list did not have an specific `class` or `id` to be select by CSS

**Comment by myl7** [comment, marked as answer]
You may do the filtering in `pages/api/index.ts` after getting all children of the folder, and the file listing view will rely on its return value.
But users can still get the original children list like via item api with the id of the folder, or via search api with the folder name as query keyword.
Anyway making a folder look empty while keeping the actual children accessible is just a visual defense.
If you are going to defend programers, a heavy work is required so maybe you need to implement them by yourself. Otherwise the above (i.e. just do filtering in `pages/api/index.ts`) should be enough.

**Reply by MuZhou233** [reply]
Thanks a lot, what I'm willing is exactly a visual defense.
